docs/jsonapi/schema.md

```diff
@@ -0,0 +1,138 @@
+| resource              | String(type), String(id),<br>attributes, relationships,<br>links, meta                                   | type, id | AM::S::Adapter::JsonApi#primary_data_for
+| links                 | Uri(self), Link(related)                                                                           |          | #1028, #1246, #1282
+| link                  | oneOf (linkString, linkObject)                                                                     |          |
+| link.linkString       | Uri                                                                                                |          |
+| link.linkObject       | Uri(href), meta                                                                                    | href     |
+| attributes            | patternProperties(<br>`"^(?!relationships$|links$)\\w[-\\w_]*$"`),<br>any valid JSON                      |          | AM::Serializer#attributes, AM::S::Adapter::JsonApi#resource_object_for
+| relationships         | patternProperties(<br>`"^\\w[-\\w_]*$"`);<br>links, relationships.data, meta                              |          | AM::S::Adapter::JsonApi#relationships_for
+| relationships.data    | oneOf (relationshipToOne, relationshipToMany)                                                      |          | AM::S::Adapter::JsonApi#resource_identifier_for
+| relationshipToOne     | anyOf(empty, linkage)                                                                              |          |
+| relationshipToMany    | UniqueArray(linkage)                                                                               |          |
+| empty                 | null                                                                                               |          |
+| linkage               | String(type), String(id), meta                                                                     | type, id | AM::S::Adapter::JsonApi#primary_data_for
+| pagination            | pageObject(first), pageObject(last),<br>pageObject(prev), pageObject(next)                            |          | AM::S::Adapter::JsonApi::PaginationLinks#serializable_hash
+| pagination.pageObject | oneOf(Uri, null)                                                                                   |          |
+| jsonapi               | String(version), meta                                                                              |          | AM::S::Adapter::JsonApi::ApiObjects::JsonApi
+| error                 | String(id), links, String(status),<br>String(code), String(title),<br>String(detail), error.source, meta |          |
+| error.source          | String(pointer), String(parameter)                                                                 |          |
+| pointer               | [JSON Pointer RFC6901](https://tools.ietf.org/html/rfc6901)                                        |          |
+
```

Comment:
reposting https://github.com/rails-api/active_model_serializers/issues/1298#issuecomment-151267763 here with a few changes, is some thoughts on why we might organize things.

let `AMS` refer to `ActiveModelSerializers`, `AM::S` to `ActiveModel::Serializer`, and `AM::S::A` to `ActiveModel::Serializer::Adapter` for brevity in the below.

| Current | Desired | Notes |
|----|---|------|
| require 'active_model_serializers' | require 'active_model_serializers'
| ActiveModelSerializers | ActiveModelSerializers 
| ActiveModel::SerializableResource | ActiveModel::SerializableResource
| namespace ActiveModel::Serializer | AMS::Current,<br>ActiveModelSerializers::V08, etc | versioned
| class AM::S  | AMS::Current::Resource | is a class that defines the properties and behavior of the data that you present to the user
|  AMS::Model |  AMS::Model | e.g. ActiveRecord object but a PORO. Maybe should be AMS::Record A record is an instance of a model that contains data loaded from a server. Your application can also create new records and save them back to the server.
| AM::S::Lint | AMS::Model::Lint
| ActiveModel::Serialization | AMS::ActionController::Serialization 
| AM::S::CollectionSerializer | AMS::Current::ResourceCollection |
| namespace AM::S::Adapter | AMS::Adapter |   adapter: knows how to talk to your server. knows how to translate requests from AMS into requests on your server.  object that translates requests from AMS/Rails/Grape etc (such as "find the user with an ID of 123") into a requests to a server.  let you completely change how your API is implemented without impacting your application code.
| class AM::S::A::JsonApi | AMS::Adapter::JsonApi,<br> AMS::Adapter::V08::Json | not versioned
| class ActiveModel::Serializer, AM::S::A::Attributes | AMS::Formatter | maps keys and values to desired format, see https://github.com/orbitjs/orbit.js/blob/master/lib/orbit-common/jsonapi-serializer.js
| N/A |  AMS::Current::Store | store is the central repository of records in your application, use the store to retrieve records, as well to create new ones. The store will automatically cache records for you. 
| N/A | AMS::Adapter::JsonApi::Exceptions 
| AM::S::A::JsonApi::ApiObject::JsonApi | AMS::Adapter::JsonApi::ApiObject::<br>{JsonApi,Links,Link,Resource,Error, etc}? vs.<br> [JsonApiFormat](https://github.com/endpoints/endpoints/blob/master/es5/format-jsonapi/index.js) or even [schema](https://github.com/orbitjs/orbit.js/blob/master/lib/orbit-common/schema.js) [validation](https://github.com/endpoints/endpoints/blob/master/es5/validate-json-schema/index.js)
| ActiveModelSerializers.logger | ActiveModelSerializers.logger 

re: `ActiveModelSerializers::Current` is just a thought.. perhaps to generate an alternative :)

see 
- http://guides.emberjs.com/v2.1.0/models/
- https://github.com/cerebris/jsonapi-resources/blob/master/lib/jsonapi/formatter.rb
- https://andycrum.github.io/ember-data-model-maker/
- https://github.com/cerebris/jsonapi-resources#jsonapiresource
- origins of naming in AMS:
  - https://github.com/rails/rails/commit/d2b78b3594b9cc9870e6a6ebfeb2e56d00e6ddb8#diff-80d5beeced9bdc24ca2b04a201543bdd
  - https://github.com/rails/rails/commit/783db25e0c640c1588732967a87d65c10fddc08e
  - https://github.com/rails/rails/commit/c6bc8e662614be711f45a8d4b231d5f993b024a7
  - https://github.com/rails/rails/commit/fbdf706fffbfb17731a1f459203d242414ef5086#diff-531b52ff1432c7c9f9472188b39d6a43
  - https://github.com/rails/rails/commit/4bfbdc133a37c6046b8854d3659ba75597d1d37e
  - https://github.com/rails/rails/commit/8896b4fdc8a543157cdf4dfc378607ebf6c10ab0
  - https://github.com/rails-api/active_model_serializers/commits/master?page=19
  - https://github.com/rails-api/active_model_serializers/commit/729a823868e8c7ac86c653fcc7100ee511e08cb6#diff-fe7aa2941c19a41ccea6e52940d84016
  - https://github.com/rails-api/active_model_serializers/commit/4a2d9853ba7486acc1747752982aa5650e7fd6e9
  - https://github.com/rails-api/active_model_serializers/commit/f00fe5595ddf741dc26127ed8fe81adad833ead5